kong/plugins/zipkin/opentracing.lua

```diff
@@ -48,23 +46,21 @@ end
 
 function OpenTracingHandler:initialise_request(conf, ctx)
 	local tracer = self:get_tracer(conf)
-	local var = ngx.var
-	local wire_context = tracer:extract("http_headers", ngx.req.get_headers()) -- could be nil
-	local method, request_url -- if reading request line fails then var.request_uri is nil
-	if var.request_uri then
-		method = ngx.req.get_method()
-		request_url = var.scheme .. "://" .. var.host .. ":" .. var.server_port .. var.request_uri
-	end
+	local req = kong.request
+	local wire_context = tracer:extract("http_headers", req.get_headers()) -- could be nil
+	local url = req.get_scheme() .. "://" .. req.get_host() .. ":"
```

Comment:
If the request was not correct, we *don't* want to use the defaults (nginx defaults method to GET, we default host and path ourselves): we want the `http.url` field to be `nil`.